File: Samples/CslaNet/vb/DeepData/DeepData.DTO/OrderDto.vb
Public Class OrderDto

  Private _id As Integer
  Public Property Id() As Integer
    Get
      Return _id
    End Get
    Set(ByVal value As Integer)
      _id = value
    End Set
  End Property

  Private _customer As String
  Public Property Customer() As String
    Get
      Return _customer
    End Get
    Set(ByVal value As String)
      _customer = value
    End Set
  End Property

  Private _lineItems As List(Of LineItemDto)
  <Xml.Serialization.XmlIgnore()> _
  Public ReadOnly Property OrderLinesList() As List(Of LineItemDto)
    Get
      If _lineItems Is Nothing Then
        _lineItems = New List(Of LineItemDto)
      End If
      Return _lineItems
    End Get
  End Property

  Public Property OrderLines() As LineItemDto()
    Get
      Return OrderLinesList.ToArray
    End Get
    Set(ByVal value As LineItemDto())
      _lineItems = New List(Of LineItemDto)(value)
    End Set
  End Property

End Class

File: Samples/CslaNet/cs/DeepData/DeepData.DTO/OrderDto.cs
using System.Collections.Generic;

namespace DeepData.DTO
{
	///<summary>
	///</summary>
	public class OrderDto
	{
		private int _id;
		///<summary>
		///</summary>
		public int Id
		{
			get
			{
				return _id;
			}
			set
			{
				_id = value;
			}
		}

		private string _customer;
		///<summary>
		///</summary>
		public string Customer
		{
			get
			{
				return _customer;
			}
			set
			{
				_customer = value;
			}
		}

		private List<LineItemDto> _lineItems;
		///<summary>
		///</summary>
		[System.Xml.Serialization.XmlIgnore()]
		public List<LineItemDto> OrderLinesList
		{
			get
			{
				if (_lineItems == null)
				{
					_lineItems = new List<LineItemDto>();
				}
				return _lineItems;
			}
		}

		///<summary>
		///</summary>
		public LineItemDto[] OrderLines
		{
			get
			{
				return OrderLinesList.ToArray();
			}
			set
			{
				_lineItems = new List<LineItemDto>(value);
			}
		}

	}
}
